doc/sphinx/proofs/automatic-tactics/logic.rst

```diff
@@ -56,7 +56,10 @@ Solvers for logic and equality
    Uses the search tree built by the decision procedure for :tacn:`tauto`
    to generate a set of subgoals equivalent to the original one (but
    simpler than it) and applies :n:`@ltac_expr` to them :cite:`Mun94`. If
-   :n:`@ltac_expr` is not specified, it defaults to :n:`auto with *`
+   :n:`@ltac_expr` is not specified, it defaults to ``Tauto.intuition_solver``.
+   The initial value of ``intuition_solver`` is equivalent to :n:`auto with *`
+   but prints warning ``intuition-auto-with-star`` when it solves a goal that
+   :tacn:`auto` cannot solve.
```

Comment:
```suggestion
   The initial value of ``intuition_solver`` is :n:`auto with *`.  With this value,
   :tacn:`tauto` prints the warning ``intuition-auto-with-star`` when it solves a goal that
   :tacn:`auto` cannot solve.
```

If the user wants `auto with *` and explictly overrides `intuition_solver`, won't they still see this message?  That seems a bit unfriendly.  Can you make the warning appear only if `intuition_solver` hasn't been changed?  Maybe something akin to:

```
let default = ref "auto with *"
let orig_default = !default
if (!default = "auto with *" && !default == orig_default) then print the warning
```

If you do this, the wording describing when the warning shows should be "if intuition_solver has been redefined" rather than "if it solves a goal with ``auto with *`` that was not solved with just :tacn:`auto`"

Reply:
>If the user wants auto with * and explictly overrides intuition_solver, won't they still see this message?

They will not, which is why your suggested change is incorrect.

Reply:
Then the description should make clear that you won't see the message if you've explicitly see `intuition_solver`.